hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/common/HoodieFlinkEngineContext.java

```diff
@@ -86,6 +110,17 @@ public RuntimeContext getRuntimeContext() {
         .collect(Collectors.toList());
   }
 
+  @Override
+  public <I, K, V> Stream<ImmutablePair<K, V>> mapPartitionsToPairAndReduceByKey(
+      Stream<I> data, SerializablePairFlatMapFunction<Iterator<I>, K, V> flatMapToPairFunc,
+      SerializableBiFunction<V, V, V> reduceFunc, int parallelism) {
+    return throwingFlatMapToPairWrapper(flatMapToPairFunc).apply(data.parallel().iterator())
+        .collect(Collectors.groupingBy(Pair::getKey)).entrySet().stream()
```

Comment:
may I know where have we used the "parallelism" within this method?

Reply:
For Flink it's not used since it's not relevant.  It's there only because of the same interface to be compatible with Spark.